sdk/core/azure-core/src/http/http.cpp

```diff
@@ -26,6 +26,7 @@ const HttpMethod HttpMethod::Post("POST");
 const HttpMethod HttpMethod::Put("PUT");
 const HttpMethod HttpMethod::Delete("DELETE");
 const HttpMethod HttpMethod::Patch("PATCH");
+const HttpMethod HttpMethod::Options("OPTIONS");
```

Comment:
An alternative stop-gap solution to adding this in core, is to just use the "OPTIONS" string. HttpMethod is an extensible enum which supports constructing from any string and is designed exactly for such scenarios.